.github/workflows/ci-cd.yml

```diff
@@ -0,0 +1,71 @@
+name: CI/CD
+
+on:
+    workflow_dispatch: # Allows you to run this workflow manually from the Actions tab
+    pull_request: # Runs whenever a pull request is created or updated
+    push: # Runs whenever a commit is pushed to the repository
+
+concurrency:
+    group: "${{ github.workflow }} @ ${{ github.event.pull_request.head.label || github.head_ref || github.ref }}"
+    cancel-in-progress: true
+
+permissions:
+    contents: write # publish a GitHub release
+    pages: write # deploy to GitHub Pages
+    issues: write # comment on released issues
+    pull-requests: write # comment on released pull requests
+
+jobs:
+    ci-cd:
+        runs-on: ubuntu-latest
+        steps:
+            - uses: actions/checkout@v4
+            - uses: wagoid/commitlint-github-action@v5
+              if: github.event_name == 'pull_request'
+            - uses: actions/setup-node@v3
+              with:
+                  cache: "npm"
+                  node-version-file: ".nvmrc"
+
+            - name: Info
+              run: |
+                  cat <<EOF
+                  Node version: $(node --version)
+                  NPM version: $(npm --version)
+                  GitHub ref: ${{ github.ref }}
+                  GitHub head ref: ${{ github.head_ref }}
+                  EOF
+
+            - name: Setup & Test
+              run: |
+                  npm ci
+                  mkdir -p ./test/results
+                  npm test
+
+            - name: Build
+              run: npm run build
+
+            - name: Generate release version
+              run: |
+                  export RELEASE_TIMESTAMP=$(date +'%Y%m%d%H%M%S')
+                  export VPKG=$($(npm bin)/json -f package.json version)
+                  export VERSION=${VPKG}-prerelease.${RELEASE_TIMESTAMP}
+
+                  echo "RELEASE_VERSION=${VERSION}" >> $GITHUB_ENV
+                  echo "NPM_TAG=latest" >> $GITHUB_ENV
```

Comment:
Looks like this is missing the hotfix handling from the CircleCI config: https://github.com/scratchfoundation/scratch-svg-renderer/blob/0705358502413eba9ab9859eed18edd4a420b23e/.circleci/config.yml#L42-L59

Sorry this is so inconsistent between repositories, but this repo in particular has been a source of hotfixes in the past so I think it's important to retain that support

Reply:
Thanks, yeah I think I ignored it for convenience :-D 
on it!